rickroll.py

```diff
@@ -153,5 +174,55 @@ def original():
 
     return str(play_tune(tune))
 
+@app.route("/sms", methods=["POST"])
+def sms():
+    """Adds an incoming SMS to a queue, to reply to later."""
+    bucket.put_object(
+        Key='queue/{to}/{from_}'.format(to=request.form['To'],
+                                        from_=request.form['From']),
+        Body='',
+    )
+    return "Hello world!"
+
+def send_sms():
+    """Empties the queue of incoming SMSes, replying to each one."""
+    for queue_entry in bucket.objects.filter(Prefix='queue/'):
+        _, our_number, their_number = queue_entry.key.split("/")
+        state_key = "state/{}/{}".format(our_number, their_number)
+
+        # Find out which message we sent last, if any
+        # Error handling taken from https://stackoverflow.com/a/33843019
+        try:
```

Comment:
Yep, no issues on separating out into a separate function. I'll do both that and what you're suggesting with error catching as well.

For logging errors from Lambda, I'd use [Sentry](https://sentry.io/welcome/); it's a really well-thought-out tool with a generous free tier, and the whole thing is open source too! (We run the open-source version on-prem at work.)